static/app/views/explore/components/chart/chartVisualization.tsx

```diff
@@ -65,23 +68,43 @@ export function ChartVisualization({
     });
   }, [chartInfo, theme]);
 
+  const previousPlottables = usePrevious(
+    plottables,
+    chartInfo.timeseriesResult.isPending
+  );
```

Comment:
How about we try `https://tanstack.com/query/v4/docs/framework/react/guides/paginated-queries#better-paginated-queries-with-keeppreviousdata`?